AutomationHelpers/src/UserCodeCollections/FileLibrary.cs

```diff
@@ -152,9 +153,169 @@ public static void WaitForFile(string path, string pattern, int duration, int in
+        /// <param name="filePath2">The relative or absolute path of the second file</param>
+        [UserCodeMethod]
+        public static void CompareBinaryFiles(string filePath1, string filePath2)
+        {
+            try
+            {
+                filePath1 = GetPathForFile(filePath1);
+                filePath2 = GetPathForFile(filePath2);
+
+                if (!FilesExist(filePath1, filePath2))
+                {
+                    return;
+                }
+
+                var fi1 = new FileInfo(filePath1);
+                var fi2 = new FileInfo(filePath2);
+
+                if (fi1.Length != fi2.Length)
+                {
+                    Report.Failure("Files '" + filePath1 + "' and '" + filePath2 + "' are not equal.");
+                    return;
+                }
+
+                int bufferSize = 65536;
+
+                using (var fs1 = fi1.OpenRead())
+                using (var fs2 = fi2.OpenRead())
+                using (var bs1 = new BufferedStream(fs1, bufferSize))
+                using (var bs2 = new BufferedStream(fs2, bufferSize))
+                {
+                    var tempBufferSize = sizeof(Int64);
+                    var tempBuffer1 = new byte[tempBufferSize];
+                    var tempBuffer2 = new byte[tempBufferSize];
+
+                    for (var i = 0L; i < (fi1.Length / tempBufferSize) + 1; i++)
+                    {
+                        bs1.Read(tempBuffer1, 0, tempBufferSize);
+                        bs2.Read(tempBuffer2, 0, tempBufferSize);
+
+                        if (BitConverter.ToInt64(tempBuffer1, 0) != BitConverter.ToInt64(tempBuffer2, 0))
+                        {
+                            Report.Failure("Files '" + filePath1 + "' and '" + filePath2 + "' are not equal.");
+                            return;
+                        }
+                    }
+                }
+
+                Report.Success("Files '" + filePath1 + "' and '" + filePath2 + "' are equal.");
+            }
+            catch (Exception ex)
+            {
+                Report.Error(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Compares content of two text files.
+        /// </summary>
+        /// <param name="filePath1">The relative or absolute path of the first file</param>
+        /// <param name="filePath2">The relative or absolute path of the second file</param>
+        [UserCodeMethod]
+        public static void CompareTextFiles(string filePath1, string filePath2)
+        {
+            try
+            {
+                filePath1 = GetPathForFile(filePath1);
+                filePath2 = GetPathForFile(filePath2);
+
+                if (!FilesExist(filePath1, filePath2))
+                {
+                    return;
+                }
+
+                var fileContent1 = File.ReadAllText(filePath1);
+                fileContent1 = Regex.Replace(fileContent1, "(\r\n)|(\n\r)|(\n)|(\r)", "\r\n");

```

Comment:
Is "\n\r" the same as "\r\n"?
Probably add an argument (and method overload) whether to normalize line endings

Reply:
Done